docs/docs/packaging.md

```diff
@@ -85,3 +85,61 @@ With conda, dependency constraints of the form `X.Y.*` will permit the installat
 pip is stricter and will not generally install nightlies unless the `--pre` argument is specified.
 However, pip can be instructed to install nightlies of specific packages by [using any specifier containing a pre-release or development version](https://pip.pypa.io/en/stable/cli/pip_install/#pre-release-versions).
 Therefore, during wheel builds we also do this [in our CI scripts by adding a `>=0.0.0a0` specifier to all the relevant constraints](https://github.com/rapidsai/cudf/blob/branch-24.06/ci/build_wheel.sh#L34).
+
+## `dependencies.yaml` Conventions
+
+Dependency sets (under the `dependencies` top-level key) shall have the
+following conventions:
+
+- Build dependencies (under `build-system.requires`) shall be under a
+  dependency set named `py_build_<wheel_name>`.
+- RAPIDS build dependencies (under `tool.rapids-build-backend.requires`) shall
+  be under a dependency set named `py_rapids_build_<wheel_name>`.
+- Runtime dependencies (under `project.dependencies`) shall be under a
+  dependency set named `py_run_<wheel_name>`.
+- Test dependencies (under `project.optional-dependencies.test`) shall be under
+  a dependency set named `py_test_<wheel_name>`.
+- CUDA-version-specific dependencies on other RAPIDS packages with a `-cu*`
+  suffix shall have a `cuda_suffixed: "true"` parameter in their matrix entry.
+  The fallback entry shall have the list of RAPIDS dependencies without their
+  `-cu*` suffix to serve as documentation in the source-controlled
+  `pyproject.toml`, and so that use cases that build the projects without the
+  `-cu*` suffix (such as DLFW) get the unsuffixed dependencies.
+- CUDA wheel dependencies (`nvidia-cublas-cu12`, etc.) shall be under a
+  dependency set named `cuda_wheels`. The CUDA-version-specific dependencies
+  (with the `-cu*` suffix) shall be under a `specific` matrix entry with
+  `cuda_version` and `use_cuda_wheels: "true"` parameters. This shall be
+  followed by a matrix entry with `use_cuda_wheels: "false"` and no packages,
+  to ensure that use cases like DLFW and devcontainers don't get CUDA wheels.
+  The fallback matrix entry shall contain all of the wheels without the `-cu*`
+  suffix. This list will never be used in an actual build, but ensures that
+  the default CUDA wheels that appear in `pyproject.toml` in source control
+  represent a CUDA-version-neutral version of the list of needed CUDA wheels.
+
+    Example:
+
+    ```
```

Comment:
```suggestion
    ```yaml
```

Reply:
I tried this but `mkdocs` didn't like it. Is there a plugin that we need?

Reply:
I don't know anything about `mkdocs`. A brief search suggests that "fenced code" is a default plugin. https://mkdocs.readthedocs.io/en/latest/user-guide/configuration/#markdown_extensions